### PR TITLE
added methods to access config and http.Client

### PR DESCRIPTION
**Pull request by fatz**
this PR is adding methods to access the config and the `http.Client` of an existing `APIClient`.

Updated the marathon example initializing a default `APIClient` and use its config and `http.Client` to configure the marathon package.

### CODE REVIEW

File: dcos/client.go [hunk 1]
```diff
@@ -101,6 +101,16 @@ func NewClientWithConfig(config *Config) (*APIClient, error) {
 	return c, nil
 }
 
+// CurrentConfig returns the *Configuration used by the client
+func (a *APIClient) CurrentConfig() *Configuration {
```
Comment (bamarni) [inline dcos/client.go]: This was not clear to me if it returns the `Configuration` or our DC/OS `Config`, does it work for your use-case to return the DC/OS config?

Right now we are in this situation with 2 config structs, as it was quickly when bootstraping the client-go code, though in the long-run we should just stick to a single config struct. So this method will likely break or its API will change in the future.

Comment (fatz) [reply dcos/client.go]: Yeah you're right... i'll try that

Comment (fatz) [reply dcos/client.go]: Oh wait.. we're not even storing the config. This is bad... 

think we need a chat about the config stuff.

I totally agree with you this should be merged into a single struct ( or maybe better interface? ) 

I mean `Configuration` totally does what I need for marathon. However I also agree with you that this is not the best situation.

The question is how to handle config changes in the users environment. @bamarni could we have a quick chat this week about Config and Configmanager? Maybe my thinking is too far but I thought of having an Config interface which could either implement a static Config or receive a more "dynamic" one via ConfigManager?

**Comment by fatz** [comment]
@bamarni i've improved the situation a bit...

so `CurrentConfig()` still gives you the APIClientConfig (Configuration)

i've added `CurrentDCOSConfig()` so a user can retrieve a copy of the dcosConfig that was being used to create the client. I especially deferenced it so noone could think of changing it while using the client.

Same for http.Client. The http.Client is also now dereferenced so the user gets a working `http.Client` but will not accidentally change the client used by APIClient

**Comment by fatz** [comment]
@bamarni dropped the `CurrentConfig()` part in favor of `CurrentDCOSConfig()`

**Pull request merged**